meson.build

```diff
@@ -0,0 +1,343 @@
+else
+    add_project_arguments('-D_TANDEM_SOURCE', language: 'c')
+    add_project_arguments('-D_ALL_SOURCE', language: 'c')
+endif
+
+lzma_c_args = []
+cflags_try = []
+
+if cc.get_argument_syntax() == 'gcc'
+    cflags_try += [
+        '-Wvla',
+        '-Wformat=2',
+        '-Winit-self',
+        '-Wmissing-include-dirs',
+        '-Wshift-overflow=2',
+        '-Wstrict-overflow=3',
+        '-Walloc-zero',
+        '-Wduplicated-cond',
+        '-Wfloat-equal',
+        '-Wundef',
+        '-Wshadow',
+        '-Wpointer-arith',
+        '-Wbad-function-cast',
+        '-Wwrite-strings',
+        '-Wdate-time',
+        '-Wsign-conversion',
+        '-Wfloat-conversion',
+        '-Wlogical-op',
+        '-Waggregate-return',
+        '-Wstrict-prototypes',
+        '-Wold-style-definition',
+        '-Wmissing-prototypes',
+        '-Wmissing-declarations',
+        '-Wredundant-decls',
+#
+        '-Wc99-compat',
+        '-Wc11-extensions',
+        '-Wc2x-compat',
+        '-Wc2x-extensions',
+        '-Wpre-c2x-compat',
+        '-Warray-bounds-pointer-arithmetic',
+        '-Wassign-enum',
+        '-Wconditional-uninitialized',
+        '-Wdocumentation',
+        '-Wduplicate-enum',
+        '-Wempty-translation-unit',
+        '-Wflexible-array-extensions',
+        '-Wmissing-variable-declarations',
+        '-Wnewline-eof',
+        '-Wshift-sign-overflow',
+        '-Wstring-conversion',
+    ]
+
+    lzma_c_args += cc.get_supported_arguments(cflags_try)
+endif
+
+config_h.set10('HAVE_VISIBILITY', cc.has_function_attribute('visibility'),
+    description: 'Define to 1 if the compiler supports simple visibility declarations.'
+)
+
+ctor_fct_src = '''
+__attribute__((__constructor__))
+static void my_constructor_func(void) { return; }
+int main() { return 0; }
+'''
+if cc.compiles(ctor_fct_src, name : 'constructor attribute for functions')
+    config_h.set10('HAVE_FUNC_ATTRIBUTE_CONSTRUCTOR', true,
+        description: 'Define to 1 if __attribute__((__constructor__)) is supported for functions.'
+    )
+endif
+
+cc.has_type('uint32_t', prefix: '#include <stdint.h>')
+
+time_fcts = [
+    [ 'futimens',  [ '#include <sys/stat.h>' ] ],
+    [ 'futimes',   [ '#include <sys/time.h>' ] ],
+    [ 'futimesat', [ '#include <fcntl.h>', '#include <sys/time.h>' ] ],
+    [ 'utimes',    [ '#include <sys/time.h>' ] ],
+    [ '_futime',   [ '#include <sys/utime.h>' ] ],
+    [ 'utime',     [ '#include <sys/types.h>', '#include <utime.h>' ] ],
+]
+foreach f : time_fcts
+    if cc.has_function(f[0], prefix : f[1])
+        config_h.set10('HAVE_' + f[0].to_upper(), true,
+            description: 'Define to 1 if the ' + f[0] + '() function is available.'
+        )
+        break
+    endif
+endforeach
+
+if cc.has_function('posix_advice', prefix : '#include <fcntl.h>')
+    config_h.set10('HAVE_POSIX_ADVICE', true,
+        description: 'Define to 1 if the posix_advice() function is available.'
+    )
```

Comment:
It's `posix_fadvise`.

Reply:
fixed in 0ffc20d22329d3227c9773d5e1c347986ff85161